mono/utils/mono-dl.c

```diff
@@ -168,6 +168,14 @@ fix_libc_name (const char *name)
 }
 #endif
 
+static int
+mono_dl_flags_external_to_internal (int flags)
+{
+	if (!(flags & MONO_DL_LOCAL))
+		flags |= MONO_DL_GLOBAL;
```

Comment:
Do you think it's worth clearing `MONO_DL_LOCAL`? not sure how embedders will behave if there are extra bits set.

Reply:
Not sure I follow. This only sets GLOBAL if LOCAL is unset, and it's a static function with its results immediately passed to `mono_dl_convert_flags` anyway so it shouldn't affect embedders.

Reply:
It's passed to the `handler->load_func` - which is the embedder, right?

But in any case, I misread the code - it sets global when local is unset.  So yea, it's ok.

Reply:
Only after `mono_dl_convert_flags`, which converts the flags over to `RTLD_*`.